Lib/defcon/objects/font.py

```diff
@@ -259,16 +264,14 @@ def _get_path(self):
 
     def _set_path(self, path):
         # XXX: this needs to be reworked for layers
+        if not isinstance(path, basestring) and not hasattr(path, "__fspath__"):
+            raise TypeError(
+                "invalid path: expected string or os.PathLike, found %s"
+                % type(path).__name__
+            )
         # the file must already exist
         assert os.path.exists(path)
-        # the glyphs directory must already exist
-        glyphsDir = os.path.join(path, "glyphs")
-        assert os.path.exists(glyphsDir)
-        # set the internal reference
         self._path = path
-        # set the glyph set reference
-        if self._glyphSet is not None:
-            self._glyphSet.dirName = glyphsDir
```

Comment:
`dirName` attribute of `glifLib.GlyphSet` is still there but it's unused, useless and deprecated.